pineappl_capi/src/lib.rs

```diff
@@ -895,6 +895,42 @@ pub unsafe extern "C" fn pineappl_grid_split_lumi(grid: *mut Grid) {
     grid.split_channels();
 }
 
+/// Change the particle ID representation of a given Grid.
+///
+/// # Safety
+///
+/// If `grid` does not point to a valid `Grid` object (for example when `grid` is the null pointer)
+/// or if `pid_basis` doe not refer to a correct basis, then this function is not safe to call.
+#[no_mangle]
+pub unsafe extern "C" fn pineappl_rotate_pid_basis(grid: *mut Grid, pid_basis: PidBasis) {
+    let grid = unsafe { &mut *grid };
+
+    grid.rotate_pid_basis(pid_basis);
+}
+
+/// Get the particle ID representation of a Grid.
+///
+/// # Safety
+///
+/// If `grid` does not point to a valid `Grid` object, for example when `grid` is the `NULL`
+/// pointer, this function is not safe to call. The parameter `key` must be non-`NULL` and a valid
+/// C string.
+///
+/// # Panics
+///
+/// This function might panic if `pid_basis` contains interior null bytes. This occurs if  a new
+/// variant is added to the `PidBasis` enum in the future and the match statement isn't updated.
+#[no_mangle]
+pub unsafe extern "C" fn pineappl_get_pid_basis(grid: *mut Grid) -> *mut c_char {
```

Comment:
This function should be called `pineappl_grid_pid_basis`. Even more importantly, I'd change the return type to `PidBasis` to mirror the behaviour of `Grid::pid_basis`. The user can then use the result and `switch`/`case` to translate the enum-value to a string if that's needed.